### DISCUSSION TITLE
Question : Unable to Search Text that includes 'Comma' or  typographical symbols using JavaScript

### DISCUSSION THREAD

**Discussion by AdityaThakor67** [supabase/supabase#19651] [Questions] [answered]
I'm working on get data on search. whenever user enter search keyword that matches with menu name or with menu_id.it works fine but when i just entered 'Comma' or  typographical symbols in search key word, i faced issue or show me error like this error object.

note : name and menu_id are field-name are string not unique.

For Example :
Search-keyword : restaurant menu ->  response - Success
Search-keyword : "restaurant,menu." ->  response - Error
 
i have tried the match() and TextSearch() but i don't wanna user enter full text to match the result if any word match the string in database
user get the data or return result the that matches the search-keyword. i have checked that using query editor if you use 
like in sql query string it work fine. so is there any way that this can be possible this kind of functionality using JavaScript.

error object :
{
    "code": "PGRST100",
    "details": "unexpected \"%\" expecting field name (* or [a..z0..9_$]), negation operator (not) or logic operator (and, or)",
    "hint": null,
    "message": "\"failed to parse logic tree ((name.ilike.%The \"Length ,%, menu_id.ilike.%The \"Length ,%))\" (line 1, column 29)"
}

```
let query = supabase
    .from("menus")
    .select(baseSelectQuery, { count: "exact" })
    .eq("restaurant_id", resId)
    .order(sort.id, { ascending: sort.desc })
    .range(offset, limit);

  // Apply search filters
  if (searchKeyword !== "") {
    query = query.or(
      `name.ilike.%${searchKeyword}%, menu_id.ilike.%${searchKeyword}%`
    );
  }

  const { error, data, count } = await query;
```

**Comment by GaryAustin1** [comment, marked as answer]
Seems like you have to escape some special characters. https://stackoverflow.com/questions/76065534/how-to-escape-string-characters-in-supabase-js-or-query

**Reply by nisarg2023** [reply]
@GaryAustin1 
I tried with double quotes but it not work with this string "The "lengthy , easdvasd ,"rdevgaerg;a'fa'e/;v,qwe'vaelgqerg"

and this string works with query = query.ilike("name", searchKeyword) 
but not with query = query.or( `name.ilike.%"${searchKeyword}"%,address.ilike.%"${searchKeyword}"%` // make sure to have proper space to work search properly);

**Reply by GaryAustin1** [reply]
I don't know.  Probably because you have " in the string itself.  Maybe need to use %22 to surround instead.  But there is also this in the source document: `If the value filtered by the in operator has a double quote ("), you can escape it using a backslash "\"". A backslash itself can be used with a double backslash "\\".` .  This seems to say you have to escape the " in the filter text with as \".  So maybe with that your call would work as is.
https://postgrest.org/en/stable/references/api/url_grammar.html#reserved-characters

 But I've never done this escaping myself.

**Comment by yaschet** [comment]
I encountered the same problem and solved it by properly encoding the filter values to handle reserved characters.

```tsx
if (pageFilter.trim()) {
    const filter = encodeURIComponent(pageFilter.trim());
    query = query.or(`display_name.ilike.%${filter}%,email.ilike.%${filter}%,phone.ilike.%${filter}%`);
}
```

**Reply by BashkirovN** [reply]
Although this saves you from the error, I believe this won't work with filters containing commas or spaces because "San Francisco" will become "San%20Francisco" after encoding and Supabase will try to match the string as is
